react/admin/CostCenterAddressList.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { useState, useEffect } from 'react'
+import { useIntl, FormattedMessage } from 'react-intl'
+import { Table, Toggle } from 'vtex.styleguide'
+
+import { costCenterMessages as messages } from './utils/messages'
+
+const tableLength = 5
+
+interface AddressListProps {
+  addressList: Address[]
+  handleNewAddressModal: () => void
+  handleCheckDefault: (address: Address) => void
+  handleEditAddressModal: (addressId: string) => void
+  handleDeleteAddressModal: (addressId: string) => void
+}
+
+const CostCenterAddressList: React.FC<AddressListProps> = ({
+  addressList,
+  handleCheckDefault,
+  handleNewAddressModal,
+  handleEditAddressModal,
+  handleDeleteAddressModal,
+}) => {
+  const { formatMessage } = useIntl()
+  const initialState = {
+    tableLength,
+    currentPage: 1,
+    filteredItems: addressList,
+    slicedData: addressList.slice(0, tableLength),
+    currentItemFrom: 1,
+    currentItemTo: tableLength,
+    searchValue: '',
+    itemsLength: addressList.length,
+    emptyStateLabel: formatMessage(messages.nothingToShow),
+    filterStatements: [],
+  }
+
+  const [state, setState] = useState(initialState)
+  const jsonschema = {
+    properties: {
+      street: {
+        title: formatMessage(messages.address),
+        cellRenderer: ({ rowData }: any) => {
+          return `${rowData.street}, ${rowData.city}, ${rowData.state}, ${rowData.postalCode}`
+        },
+      },
+      receiverName: {
+        title: formatMessage(messages.receiverName),
+        cellRenderer: ({ rowData }: any) => {
+          try {
+            const receiver = JSON.parse(rowData?.receiverName)
+            const strings = Object.keys(receiver).map(
+              (key: string) => receiver[key]
+            )
+
+            return strings.join(' - ')
+          } catch (e) {
+            return rowData?.receiverName
+          }
+        },
+      },
+      checked: {
+        title: formatMessage(messages.defaultAddress),
+        width: 150,
+        cellRenderer: ({ rowData }: any) => {
+          return (
+            <Toggle
+              onChange={() => {
+                handleCheckDefault(rowData)
+              }}
+              checked={rowData.checked}
+            />
+          )
+        },
+      },
+    },
+  }
+
+  useEffect(() => {
+    if (addressList.length > 0) {
```

Comment:
It might be interesting to check if `addressList` is null here

```suggestion
    if (addressList && addressList.length > 0) {
```